tcms/core/tests/test_ajax.py

```diff
@@ -35,6 +44,48 @@ def test_empty_string_is_converted_to_bool(self):
         response = self.client.get(url)
         self.assertEqual(200, response.status_code)
 
+    def test_with_ulli_format(self):
+        """ When a request comes with info_type=categories, and format=ulli for given product_id,
+            we expect to receive all categories for that product, rendered as HTML unordered list
+        """
+
+        url = '%s?info_type=categories&product_id=%d&format=ulli' % (reverse('ajax-info'),
+                                                                     self.product.pk)
+
+        response = self.client.get(url)
+
+        self.assertContains(response, '<li>' + self.category_one.name + '</li>')
+        self.assertContains(response, '<li>' + self.category_two.name + '</li>')
+
+    def test_with_unrecognisable_infotype(self):
+        """ When a request comes with invalid info_type,
+            we expect to receive response containing the 'Unrecongnizable infotype' error message
+        """
+
+        url = "%s?info_type=INVALID" % reverse('ajax-info')
+
+        response = self.client.get(url)
+
+        self.assertEqual(str(response.content, encoding=settings.DEFAULT_CHARSET),
+                         'Unrecognizable infotype')
+
+    def test_with_json_format(self):
+        """ When a request comes with info_type=categories for given product_id,
+            we expect to receive all categories for that product as array of JSON objects """
+
+        url = "%s?info_type=categories&product_id=%d" % (reverse('ajax-info'), self.product.pk)
+
+        response = self.client.get(url)
+
+        self.assertIn(
+            json.loads('{"model": "testcases.category", "pk": %s, "fields": {"name": "%s"}}'
+                       % (self.category_one.pk, self.category_one.name)),
+            json.loads(response.content, encoding=settings.DEFAULT_CHARSET))
+        self.assertIn(
+            json.loads('{"model": "testcases.category", "pk": %s, "fields": {"name": "%s"}}'
+                       % (self.category_two.pk, self.category_two.name)),
+            json.loads(response.content, encoding=settings.DEFAULT_CHARSET))
```

Comment:
The reason assertIn works is the following:

```
expected = {'fields': {'name': 'category 0'}, 'model': 'testcases.category', 'pk': 2}

actual = [
 {'fields': {'name': '--default--'}, 'model': 'testcases.category', 'pk': 1},
 {'fields': {'name': 'category 0'}, 'model': 'testcases.category', 'pk': 2},
 {'fields': {'name': 'category 1'}, 'model': 'testcases.category', 'pk': 3}
]
```

So a member is always **in** a list. You can keep the asserts but a few notes:

1) assign the resulting value to `actual` or something like that to make things clear and also avoid decoding and converting the response to JSON twice
2) Add an assertion for the `-- default --` value.